connector/github/github.go

```diff
@@ -411,12 +416,89 @@ func (c *githubConnector) user(ctx context.Context, client *http.Client) (user,
 		return u, fmt.Errorf("%s: %s", resp.Status, body)
 	}
 
-	if err := json.NewDecoder(resp.Body).Decode(&u); err != nil {
+	if err = json.NewDecoder(resp.Body).Decode(&u); err != nil {
 		return u, fmt.Errorf("failed to decode response: %v", err)
 	}
+
+	if u.Email == "" {
+		if u.Email, err = c.userEmail(ctx, client); err != nil {
+			return u, err
+		}
+	}
 	return u, nil
 }
 
+type userEmail struct {
+	Email      string `json:"email"`
+	Verified   bool   `json:"verified"`
+	Primary    bool   `json:"primary"`
+	Visibility string `json:"visibility"`
+}
+
+// userEmail queries the GitHub API for a users' email information using the
+// provided client. Only returns the users' verified, primary email (private or
+// public).
+//
+// The HTTP client is expected to be constructed by the golang.org/x/oauth2 package,
+// which inserts a bearer token as part of the request.
+func (c *githubConnector) userEmail(ctx context.Context, client *http.Client) (string, error) {
+
+	// https://developer.github.com/v3/#pagination
+	reNext := regexp.MustCompile("<(.*)>; rel=\"next\"")
+	reLast := regexp.MustCompile("<(.*)>; rel=\"last\"")
+	// https://developer.github.com/v3/users/emails/#list-email-addresses-for-a-user
+	apiURL := c.apiURL + "/user/emails"
+
+	var emails []userEmail
+	for {
+		req, err := http.NewRequest("GET", apiURL, nil)
```

Comment:
Split this GET, handle bad response, deserialize response body into a helper and share it with the method above.

```go
func get(ctx context.Context, client *http.Client, apiURL, path string, resp interface{})
``` 

This method then becomes something like

```go
    var emails []userEmail
    if err := get(ctx, client, apiURL, "/user/emails", &emails); err != nil {
    }
```